ports/stm32/mboot/main.c

```diff
@@ -464,31 +466,37 @@ static int usrbtn_state(void) {
 #endif
 
 static int mboot_flash_mass_erase(void) {
-    // TODO
+    #if defined(STM32F4) || defined(STM32F7) || defined(STM32H7)
+    // Erase all flash pages after mboot.
+    int ret = flash_erase(APPLICATION_ADDR, APPLICATION_FLASH_LENGTH / sizeof(uint32_t));
+    return ret;
+    #else
     return -1;
+    #endif
 }
 
 static int mboot_flash_page_erase(uint32_t addr, uint32_t *next_addr) {
     uint32_t sector_size = 0;
-    uint32_t sector = flash_get_sector_info(addr, NULL, &sector_size);
-    if (sector == 0) {
-        // Don't allow to erase the sector with this bootloader in it
+    uint32_t sector_start = 0;
+    int32_t sector = flash_get_sector_info(addr, &sector_start, &sector_size);
```

Comment:
`flash_get_sector_info` returns an unsigned value, and 0 for invalid address... I guess that function should be changed to return -1 on error

Reply:
Ah yes, I'd made a similar change to the equivalent local copy of that function previously.